npm/pod.go

```diff
@@ -35,6 +36,20 @@ func (npMgr *NetworkPolicyManager) AddPod(podObj *corev1.Pod) error {
 	podIP := podObj.Status.PodIP
 	log.Printf("POD CREATING: [%s/%s/%s%+v%s]", podNs, podName, podNodeName, podLabels, podIP)
 
+	// Add pod namespace if it doesn't exist
+	if _, exists := npMgr.nsMap[podNs]; !exists {
+		nsObj := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podObj.ObjectMeta.Namespace,
+				Labels: make(map[string]string),
+			},
+		}
+		
+		if err = npMgr.AddNamespace(nsObj); err != nil {
```

Comment:
Do you know why it worked without this change previously? The functions are locked so there shouldn't be any race conditions...

Reply:
it's only when they have a namespace that hasn't been registered yet

so if sharedinformercache triggers the addnetworkpolicy or addnode call before addnamespace, there will be a failure (on reboot)